lib/stm32/common/usart_common_all.c

```diff
@@ -94,6 +94,20 @@ void usart_set_baudrate(uint32_t usart, uint32_t baud)
 #endif
 }
 
+/*---------------------------------------------------------------------------*/
+/** @brief USART Get Baudrate.
+
+@param[in] usart unsigned 32 bit. USART block register address base @ref usart_reg_base
+@returns baud unsigned 32 bit. Baud rate specified in Hz.
+*/
+
+uint32_t usart_get_baudrate(uint32_t usart)
+{
+	uint32_t clock = rcc_get_usart_clk_freq(usart);
+	const uint32_t reg32 = USART_BRR(usart);
+	return clock / reg32;
```

Comment:
Added untested code. Should I also handle LPUART1 specially?

Reply:
Yes, otherwise these are latent bugs due to the mismatched implementations. We would suggest handling LPUART1 is probably a smart/necessary idea. We'll test the OVER8 code on a STM32H7 with known PLL setup and dividers so we know what we should expect out the calculation.

Reply:
Added LPUART1 block as-if simple 256x clock. 
The more I look at it, the more I think it should go into usart_common_f124.c (F1 no OVER8, F2, F4, L1) or even usart_common_v2.c (F0, F3, F7, G0, G4, H7, L0, L4). Were there any specific reasons why you duplicated your `usart_set_oversampling()` to both sources? Should I? (I originally aimed at F4 only, but the logic for others could be shared, like F723 in `stlinkv3`)

Reply:
We did it that way to match the organisation of other similar functions in the code, and so we'd have access to that LIBOPENCM3_USART_COMMON_F24_H macro to guard it's definition on F1 parts.